### DISCUSSION TITLE
More details for discovered integrations

### DISCUSSION THREAD

**Discussion by postlund** [home-assistant/frontend#8555] [Configuration] [answered]
So, this happened when I upgraded to the latest release yesterday:

![CDB48914-0196-40DC-96F7-F26216E62FE0](https://user-images.githubusercontent.com/5982179/110086885-3b179d80-7d93-11eb-930b-32eada530c86.jpeg)

I did not know I have a Xiaomii device, nor do I have any way to identify which device it is as no device specific device is available here. Ideally the miio integration would supply a name or something from the discovery info (and that would make it a "bug" in that integration). But I do feel it would make sense to have a small info button in the corner with generic information, like the IP address or even a raw dump of the discovery info (at least in advanced mode). I got a related issue myself, where the Apple TV integration would discover a device that is not an Apple TV but that uses the same protocols. Even with device name, it's not always way to realize which device that is.

**Comment by frenck** [comment, marked as answer]
This is not a frontend implementation issue, as this is already possible. It is up to the integration to provide this information.
Therefore, there is nothing to do with this request.

**Reply by postlund** [reply]
Yeah, the integrations *can* provide some information (like name) but there's no rules enforcing that during review, right? It would be very convenient if the frontend could at least display the IP address of the device as a help, when the integration does not provide any helpful identifiers?

**Reply by frenck** [reply]
> Yeah, the integrations can provide some information (like name) but there's no rules enforcing that during review, right?

It all depends on the integration, not all can provide information; others can only have a single instance so in that case it doesn't matter either. There is not golden standard on this.

Either way, this is not a frontend issue. The frontend is capable of displaying the information, however, if the information is not presented by the integration then there is nothing the frontend can do.

**Reply by postlund** [reply]
You are correct of course, but I'm only talking about integrations discovered via the discovery API, like Zeroconf, SSDP, HomeKit, etc. All those devices are discovered on the network, no? So they should at least have an IP address of some sort associated with them. MQTT would be a bad example in this case, but the others should be ok I think. I do understand that it's not an issue per se with the frontend now, as there's no support in backend for this (which could be added I guess). It's more like a feature request as it's a bit annoying to not know what an integration actually "is" in this case.

**Reply by frenck** [reply]
In that case, it would still be a backend issue, not a frontend one (where you are currently trying to discuss this). As said, the frontend will display the data; if provided.

**Reply by postlund** [reply]
Ok, then I will move the discussion to backend instead. Cheers! 🍻
